api/tests/opentrons/protocols/geometry/test_deck.py

```diff
@@ -0,0 +1,119 @@
+"""Tests for the opentrons.protocols.geometry.Deck interface."""
+from typing import Any
+
+import pytest
+from decoy import Decoy
+
+from opentrons.hardware_control.modules import ModuleType
+from opentrons.protocols.geometry import deck_conflict
+from opentrons.protocols.geometry.deck import Deck
+from opentrons.protocols.geometry.deck_item import DeckItem
+from opentrons.protocols.geometry.module_geometry import ModuleGeometry
+
+
+@pytest.fixture(autouse=True)
+def mock_deck_conflict_check(decoy: Decoy, monkeypatch: pytest.MonkeyPatch) -> None:
+    """Mock out the deck conflict checker."""
+    mock_check = decoy.mock(func=deck_conflict.check)
+    monkeypatch.setattr(deck_conflict, "check", mock_check)
+
+
+@pytest.fixture
+def subject() -> Deck:
+    """Get a Deck test subject."""
+    return Deck()
+
+
+@pytest.mark.parametrize("slot_number", range(1, 12))
+def test_slot_names(decoy: Decoy, slot_number: int, subject: Deck) -> None:
+    """Ensure item get/set/delete works with both strs and ints."""
+    deck_item = decoy.mock(cls=DeckItem)
+    decoy.when(deck_item.highest_z).then_return(42)
+
+    slot_name = str(slot_number)
+
+    subject[slot_number] = deck_item
+    assert subject[slot_number] is deck_item
+    assert subject[slot_name] is deck_item
+
+    del subject[slot_number]
+    assert subject[slot_number] is None
+    assert subject[slot_name] is None
+
+    subject[slot_name] = deck_item
+    assert subject[slot_number] is deck_item
+    assert subject[slot_name] is deck_item
+
+    del subject[slot_name]
+    assert subject[slot_number] is None
+    assert subject[slot_name] is None
+
+
+@pytest.mark.parametrize("bad_slot", [0, 13, "0", "13", "hello world"])
```

Comment:
```suggestion
@pytest.mark.parametrize("bad_slot", [0, 13, 1.0, "0", "13", "1.0", "hello world"])
```

A couple more interesting test cases to throw in, if you want.